### PR TITLE
Add vLLM Support for GPT OSS and its mapping generator for tunix.

### PR DESCRIPTION
**Pull request by abhinavclemson**
# Description
#### Add vLLM Support for GPT OSS and its mapping generator for tunix.

Before submitting this PR, please make sure (put X in square brackets):

(https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).kets):
- [X] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run end-to-end tests tests and provided workload links above if applicable.
- [X] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/MaxText/integration/tunix/weight_mapping/gpt_oss.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+      # Safety Check
+      if current.shape[-1] != val.shape[-1] * 2:
+        if current.shape[-1] == val.shape[-1]:
+          logging.debug("Gate Fusion Shape Warning: Src %s -> Tgt %s", val.shape, current.shape)
+          return val
+        logging.warning("Gate Fusion Shape Warning: Src %s -> Tgt %s", val.shape, current.shape)
+
+      # TODO: Enable multi-host sharding, if there is a mismatch in shapes.
+      # # MULTI-HOST case.
+      val = jax.device_put(val, current.sharding)
```
Comment (gagika) [inline src/MaxText/integration/tunix/weight_mapping/gpt_oss.py:59]: was it tested at multi-host ?

what sharding do you think it might not work?

File: src/MaxText/integration/tunix/weight_mapping/gpt_oss.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+      val.block_until_ready()
+
+      logging.debug("Hook: Interleaving Gate -> Even columns")
+      return current.at[..., 0::2].set(val)
+
+    def fuse_interleaved_up(val, tgt_param):
+      """Fuse Up (wi_1) with Multi-Host Sharding Support."""
```
Comment (gagika) [inline src/MaxText/integration/tunix/weight_mapping/gpt_oss.py:66]: why `Multi-Host` is special? do you mean multi-devie or multi-host?

From Jax's perspective I think only number of devices (and sharding) matters.

File: src/MaxText/integration/tunix/utils.py [hunk 3]
```diff
@@ -127,7 +129,17 @@ def __init__(self, model_name, config=None, use_standalone_mappings=False):
   def to_hf_mapping(self):
     """Returns a mapping from MaxText parameter names to HuggingFace parameter names."""
     if self.use_standalone_mappings:
-      return STANDALONE_VLLM_WEIGHT_MAPPING[self.model_name].to_hf_mapping()
+      mapping_fn = STANDALONE_VLLM_WEIGHT_MAPPING[self.model_name].to_hf_mapping
+      total_num_layers = self.config["num_hidden_layers"]
+      print(f"total_num_layers: {total_num_layers} for model: {self.model_name}")
```
Comment (gagika) [inline src/MaxText/integration/tunix/utils.py:134]: could you remove this print or make debug logging  (e.g. logging.debug)?

File: src/MaxText/integration/tunix/weight_mapping/gpt_oss.py [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+      if current.shape[-1] != val.shape[-1] * 2:
+        if current.shape[-1] == val.shape[-1]:
+          logging.debug("Gate Fusion Shape Warning: Src %s -> Tgt %s", val.shape, current.shape)
+          return val
+        logging.warning("Gate Fusion Shape Warning: Src %s -> Tgt %s", val.shape, current.shape)
+
+      # TODO: Enable multi-host sharding, if there is a mismatch in shapes.
+      # # MULTI-HOST case.
+      val = jax.device_put(val, current.sharding)
+      val.block_until_ready()
```
Comment (gagika) [inline src/MaxText/integration/tunix/weight_mapping/gpt_oss.py:60]: is this needed?
